### PR TITLE
Add SQL summary edge cases tests

### PR DESCRIPTION
**Pull request by trask**

### CODE REVIEW

File: instrumentation-api-incubator/src/test/java/io/opentelemetry/instrumentation/api/incubator/semconv/db/SqlQuerySummaryEdgeCasesTest.java [hunk 1]
```diff
@@ -0,0 +1,817 @@
...
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/**
+ * AI generated edge case tests for SQL query summary generation. Covers database-specific syntax,
+ * subqueries, CTEs, window functions, vendor-specific features, and various lexical edge cases.
+ */
+class SqlQuerySummaryEdgeCasesTest {
```
Comment (trask) [inline instrumentation-api-incubator/src/test/java/io/opentelemetry/instrumentation/api/incubator/semconv/db/SqlQuerySummaryEdgeCasesTest.java:20]: @laurit what do you think about checking this in? I've been using it locally to help find and fix edge cases in the sql summarization. not sure if it's worth baking in to the main test, or leaving as "edge cases" test, or just dropping it

Comment (laurit) [reply instrumentation-api-incubator/src/test/java/io/opentelemetry/instrumentation/api/incubator/semconv/db/SqlQuerySummaryEdgeCasesTest.java:20]: Let's add it.
